cmd/webservice/main.go

```diff
@@ -33,13 +37,40 @@ func getConfig() *WSConfig {
 	return _config
 }
 
-func getRsa(paas string) *crypt.Crypt {
-	if _crypt == nil {
-		_crypt = make(map[string]*crypt.Crypt)
+func resetRsa() {
+	log.Println("Resetting RSA")
+	_cryptLock.Lock()
+	defer _cryptLock.Unlock()
+	_crypt = make(map[string]*crypt.Crypt)
+}
+
+func getCrypt(paas string) *crypt.Crypt {
+	_cryptLock.RLock()
+	defer _cryptLock.RUnlock()
+	if c, exists := _crypt[paas]; exists {
+		return c
 	}
+	return nil
+}
 
+func getRsa(paas string) *crypt.Crypt {
 	config := getConfig()
-	if c, exists := _crypt[paas]; exists {
+	if _fw == nil {
+		log.Println("Starting watcher")
+		_fw = utils.NewFileWatcher(config.PrivateKeyPath, config.PublicKeyPath)
+	}
+	// It is crucial that we have this first and nil check on _crypt later
+	if _fw.WasTriggered() {
+		log.Println("Files changed")
+		resetRsa()
+	} else if _crypt == nil {
+		log.Println("crypt empty")
+		resetRsa()
+	}
+	if _crypt == nil {
+		log.Println("crypt empty2")
+	}
```

Comment:
```suggestion
	if _fw.WasTriggered() {
		log.Println("Files changed")
		resetRsa()
	}
```
I don't know if I'm correct, but 66-72 feels duplicated. And how can crypt ever be nil as it is initialised as a map in it's var?

Reply:
Yes, I added this for debugging g purposes and left it by mistake.
Removed...

Reply:
Can you specify why the `_crypt == nil` check is here? IMO this is never the case. That's why I left that lines out of my suggestion.

Reply:
When the webservice starts, main never initializes _crypt.
When an api call does something with encrypten, getRsa is the first touchpoint with _crypt, which initially finds it uninitialized.
So, mentioned check finds that out and initializes it by running resetRsa().
Without this if block, a nilpointer exception is raised on first api call that does something with encryption...